neural_style.py

```diff
@@ -155,10 +157,22 @@ def main():
                      "parameter must contain `%s` (e.g. `foo%s.jpg`)")
 
     # try saving a dummy image to the output path to make sure that it's writable
+    overwrite_error = False
     try:
-        imsave(options.output, np.zeros((500, 500, 3)))
-    except:
-        raise IOError('%s is not writable or does not have a valid file extension for an image file' % options.output)
+        if not options.overwrite:
+            if not os.path.isfile(options.output):
+                imsave(options.output, np.zeros((500, 500, 3)))
+            else:
+                overwrite_error = True
+                raise IOError
+        else:
+            # Overwrite
+            imsave(options.output, np.zeros((500, 500, 3)))
+    except IOError:
+        if overwrite_error:
+            raise IOError("%s already exists, will not replace it without the '--overwrite' flag" % options.output)
+        else:
+            raise IOError('%s is not writable or does not have a valid file extension for an image file' % options.output)
```

Comment:
The logic in this block of code can be simplified quite a bit, and without repetition of the `imsave()` / without catching and re-throwing the overwrite error. The check can be lifted outside the `try/except`. E.g.:

```python
if os.path.isfile(options.output) and not options.overwrite:
    raise IOError('%s already exists, will not replace it without the "--overwrite" flag' % options.output)
try:
    imsave(options.output, np.zeros((500, 500, 3)))
except:
    raise IOError('%s is not writable or does not have a valid file extension for an image file' % options.output)
```

This would be a much smaller diff (2 lines added) to the old version of the code.